crates/shared/src/price_estimation/native/coingecko.rs

```diff
@@ -90,7 +102,12 @@ impl CoinGecko {
         if !response.status().is_success() {
             let status = response.status();
             return match status {
-                StatusCode::TOO_MANY_REQUESTS => Err(PriceEstimationError::RateLimited),
+                StatusCode::TOO_MANY_REQUESTS => {
+                    return Ok(tokens
+                        .iter()
+                        .map(|token| (**token, Err(PriceEstimationError::RateLimited)))
```

Comment:
Why do you return a rate limiting error for every token here instead of one single `Err` from the function?
That seems to describe better what kind of error actually happened. 🤔